From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: s911415 <s911415@users.noreply.github.com>
Date: Sun, 29 Mar 2020 18:04:15 +0800
Subject: cherry-pick: Migrate Deprecated Features for PHP 7.4

In lib.misc.php
Fix nested ternary operators without explicit parentheses
Fix array and string offset access using curly braces

(cherry picked from commit c79137de07e1b3eb9c2398a8f9ea41ee2f9613df)
---
 lib/lib.misc.php | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/lib.misc.php b/lib/lib.misc.php
index f2bb8051f..926900138 100644
--- a/lib/lib.misc.php
+++ b/lib/lib.misc.php
@@ -772,7 +772,7 @@ function eventlog(string $type, $dataids, string $action, $cid = null, $json = n
 
     // Make a combined string to keep track of the data ID's
     $dataidsCombined = json_encode($dataids);
-    $idsCombined = $ids === null ? null : is_array($ids) ? json_encode($ids) : $ids;
+    $idsCombined = $ids === null ? null : (is_array($ids) ? json_encode($ids) : $ids);
 
     logmsg(LOG_DEBUG, "eventlog arguments: '$type' '$dataidsCombined' '$action' '$cid' '$json' '$idsCombined'");
 
@@ -968,7 +968,7 @@ function read_API_credentials()
         error("Cannot read REST API credentials file " . $credfile);
     }
     foreach ($credentials as $credential) {
-        if ($credential{0} == '#') {
+        if ($credential[0] == '#') {
             continue;
         }
         list($endpointID, $resturl, $restuser, $restpass) = preg_split("/\s+/", trim($credential));
-- 
2.20.1

